.github/workflows/CI-lint-format.yml

```diff
@@ -43,8 +39,24 @@ jobs:
             rustup target add wasm32-unknown-unknown
             rustup component add rust-src
       - name: Linting
-        if: matrix.os == 'ubuntu-latest'
-        run: cargo clippy --release -- --deny warnings
+        run: cargo clippy --release -- --deny warnings 2>&1 | tee linting_output.txt
+      - name: Upload Linting Output
+        uses: actions/upload-artifact@v4
+        with:
+          name: linting-output
+          path: linting_output.txt
+          if-no-files-found: warn
+          retention-days: 1
+          compression-level: 0
+          overwrite: true
       - name: Formatting
-        if: matrix.os == 'ubuntu-latest'
-        run: cargo fmt --check
\ No newline at end of file
+        run: cargo fmt -- --check 2>&1 | tee formatting_output.txt
```

Comment:
Is that additional `--` introduced on purpose or is it a leftover? It seems to me no big difference btw.

Reply:
Good spot i've removed that additional --